### ISSUE TITLE
glError 1280: GL_INVALID_ENUM in glDrawElements

### ISSUE THREAD

**Issue by sdelaysam** [fabmax/LightGL#1]
When ObjLoader loads model, it writes mesh indices to IntBuffer.
Mesh creates index buffer with type GL_UNSIGNED_INT.
Method glDrawElements crashes with GL_INVALID_ENUM on some devices.
http://www.gsmarena.com/samsung_galaxy_ace_nxt-6534.php

According to https://www.khronos.org/opengles/sdk/docs/man/xhtml/glDrawElements.xml  GL_INVALID_ENUM is generated if type is not GL_UNSIGNED_BYTE or GL_UNSIGNED_SHORT.

Tried to change IntBuffer to ShortBuffer in ObjLoader.loadObj and it worked.
Maybe, there is more reasonable way to fix it?

**Comment by fabmax** [comment]
No, changing the IntBuffer to a ShortBuffer is the way to go. I originally wrote that code for desktop OpenGL where GL_UNSIGNED_INT is fine and the phones I tested did support it as well, but to be safe it should use GL_UNSIGNED_SHORT.
